src/cluster_linearize.h

```diff
@@ -962,6 +962,47 @@ void PostLinearize(const DepGraph<SetType>& depgraph, Span<ClusterIndex> lineari
     }
 }
 
+/** Merge two linearizations for the same cluster into one that is as good as both.
+ *
+ * Complexity: O(N^2) where N=depgraph.TxCount(); O(N) if both inputs are identical.
+ */
+template<typename SetType>
+std::vector<ClusterIndex> MergeLinearizations(const DepGraph<SetType>& depgraph, Span<const ClusterIndex> lin1, Span<const ClusterIndex> lin2)
+{
+    Assume(lin1.size() == depgraph.TxCount());
+    Assume(lin2.size() == depgraph.TxCount());
+
+    /** Chunkings of what remains of both input linearizations. */
+    LinearizationChunking chunking1(depgraph, lin1), chunking2(depgraph, lin2);
+    /** Output linearization. */
+    std::vector<ClusterIndex> ret;
+    if (depgraph.TxCount() == 0) return ret;
+    ret.reserve(depgraph.TxCount());
+
+    while (true) {
+        // As long as we are not done, both linearizations must have chunks left.
+        Assume(chunking1.NumChunksLeft() > 0);
+        Assume(chunking2.NumChunksLeft() > 0);
+        // Find the set to output by taking the best remaining chunk, and then intersecting it with
+        // prefixes of remaining chunks of the other linearization.
+        SetInfo<SetType> best;
+        const auto& lin1_firstchunk = chunking1.GetChunk(0);
+        const auto& lin2_firstchunk = chunking2.GetChunk(0);
+        if (lin2_firstchunk.feerate >> lin1_firstchunk.feerate) {
+            best = chunking1.Intersect(lin2_firstchunk);
```

Comment:
hm, `Intersect` had me quite confused in this context, consider renaming this as something like `PrefixIntersect`?